app/lib/features/chat/providers/notifiers/media_chat_notifier.dart

```diff
@@ -33,14 +33,21 @@ class MediaChatNotifier extends StateNotifier<MediaChatState> {
       );
       try {
         //Get media path if already downloaded
-        final mediaPath = await _convo!.mediaPath(messageInfo.messageId, false);
-        if (mediaPath.text() != null) {
-          final videoThumbnailFile = await getThumbnailData(mediaPath.text()!);
+        final mediaPath =
+            (await _convo!.mediaPath(messageInfo.messageId, false)).text();
+
+        if (mediaPath != null) {
           state = state.copyWith(
-            mediaFile: File(mediaPath.text()!),
-            videoThumbnailFile: videoThumbnailFile,
+            mediaFile: File(mediaPath),
+            videoThumbnailFile: null,
             mediaChatLoadingState: const MediaChatLoadingState.loaded(),
           );
+          final videoThumbnailFile = await getThumbnailData(mediaPath);
+          if (videoThumbnailFile != null) {
+            if (state.mediaFile?.path == mediaPath) {
+              state = state.copyWith(videoThumbnailFile: videoThumbnailFile);
+            }
+          }
```

Comment:
Assuming that this changes are for updating change in 2 separate action for file and thumb so that one can not have to wait.